### PR TITLE
[IR] Remove zext and sext constant expressions

### PR DESCRIPTION
**Pull request by nikic**
Remove support for zext and sext constant expressions. All places creating them have been removed beforehand, so this just removes the APIs and uses of these constant expressions in tests.

There is some additional cleanup that can be done on top of this, e.g. we can remove the ZExtInst vs ZExtOperator footgun.

This is part of https://discourse.llvm.org/t/rfc-remove-most-constant-expressions/63179.

### CODE REVIEW

**Comment by jayfoad** [comment]
Remove them from https://llvm.org/docs/LangRef.html#constant-expressions?

**Comment by nikic** [comment]
> Remove them from https://llvm.org/docs/LangRef.html#constant-expressions?

Done!

File: llvm/lib/IR/ConstantFold.cpp [hunk 1]
```diff
@@ -1461,17 +1411,13 @@ static ICmpInst::Predicate evaluateICmpRelation(Constant *V1, Constant *V2,
       [[fallthrough]];
     case Instruction::UIToFP:
     case Instruction::SIToFP:
-    case Instruction::ZExt:
```
Comment (nhaehnle) [inline llvm/lib/IR/ConstantFold.cpp:1464]: This code puzzles me. Not so much your change to it, but...

The method is called `evaluateICmpRelation`, and a whole lot of comments talk about how we can't evaluate floating point values. So how can we ever get the `UIToFP` or `SIToFP` cases here?

**Review by nhaehnle — APPROVED** [review]
Thank you for doing this. I looked through the code changes and a large random sample of the test changes, and they all look good to me except one spot where I'm confused -- and I suppose the change is fine as a matter of doing a mechanical change, but perhaps that code could be cleaned up further? Though if I'm right about it, it may be for a follow-up commit anyway.

File: llvm/lib/IR/ConstantFold.cpp (hunk 1 above)
Comment (nikic) [reply llvm/lib/IR/ConstantFold.cpp:1464]: I agree that this doesn't make sense. I'll clean it up in a followup change.

**Comment by lattner** [comment]
awesome!!

**Pull request merged**